src/yachtzee/diet.ts

```diff
@@ -0,0 +1,692 @@
+      throw new Error(
+        `Error in diet schedule: Overeating ${entry.quantity} ${entry.name} to ${fullness}/${
+          fullnessLimit() + toInt(haveDistentionPill)
+        }`
+      );
+    } else if (drunkenness > inebrietyLimit()) {
+      throw new Error(
+        `Error in diet schedule: Overdrinking ${entry.quantity} ${
+          entry.name
+        } to ${drunkenness}/${inebrietyLimit()}`
+      );
+    } else if (spleenUse > spleenLimit()) {
+      throw new Error(
+        `Error in diet schedule: Overspleening ${entry.quantity} ${
+          entry.name
+        } to ${spleenUse}/${spleenLimit()}`
+      );
+    }
+  }
+
+  return dietSchedule;
+}
+
+export function yachtzeeChainDiet(simOnly?: boolean): boolean {
+  if (get("_garboYachtzeeChainDietPlanned", false)) return true;
+
+  const havePYECCharge = pyecAvailable();
+  const haveDistentionPill = !get("_distentionPillUsed") && have($item`distention pill`);
+  const haveDogHairPill = !get("_syntheticDogHairPillUsed") && have($item`synthetic dog hair pill`);
+
+  const currentSpleenLeft = spleenLimit() - mySpleenUse();
+  const filters = 3 - get("currentMojoFilters");
+  // save some spleen the first two extro, which are worth a lot
+  // due to macrometeor and cheat code: replace enemy
+  const extroSpleenSpace = hasMonsterReplacers()
+    ? 4 - Math.min(4, 2 * get("beGregariousCharges"))
+    : 0;
+  const synthCastsToCoverRun =
+    globalOptions.noBarf || !have($skill`Sweet Synthesis`)
+      ? 0
+      : Math.max(0, Math.round((estimatedTurns() - haveEffect($effect`Synthesis: Greed`)) / 30));
+  const fullnessAvailable = fullnessLimit() - myFullness() + toInt(haveDistentionPill);
+  const inebrietyAvailable =
+    inebrietyLimit() -
+    myInebriety() +
+    toInt(haveDogHairPill) +
+    (3 - get("_sweatOutSomeBoozeUsed", 0));
+  const spleenAvailable = currentSpleenLeft + filters;
+  const organsAvailable = fullnessAvailable + inebrietyAvailable + spleenAvailable;
+
+  const cleanableSpleen = organsAvailable - synthCastsToCoverRun - extroSpleenSpace;
+  const sufficientOrgansFor = (yachtzees: number) =>
+    cleanableSpleen >= yachtzees + (havePYECCharge ? 5 : 0);
+
+  const possibleBaseYachtzeeTurns = [30, 20, 10];
+  const baseYachtzeeTurns = possibleBaseYachtzeeTurns.find(sufficientOrgansFor) ?? 0;
+
+  if (baseYachtzeeTurns === 0) {
+    print("Determined that there are no suitable number of turns to chain yachtzees", "red");
+    return false;
+  }
+
+  const maxYachtzeeTurns = havePYECCharge ? baseYachtzeeTurns + 5 : baseYachtzeeTurns;
+  print(`Synth Casts Wanted: ${synthCastsToCoverRun}`, "blue");
+  print(`Organs Available: ${organsAvailable}`, "blue");
+  print(`Max Yachtzee Turns: ${maxYachtzeeTurns}`, "blue");
+
+  // Plan our diet
+
+  const sliders = Math.floor((fullnessLimit() + toInt(haveDistentionPill) - myFullness()) / 5);
+  const pickleJuice = Math.floor((inebrietyLimit() - myInebriety()) / 5);
```

Comment:
Once we support sweat out and synthetic pill, we have to add their contributions here (just like the `haveDistentionPill`) contribution to sliders above

Reply:
Out of scope of this PR